ui/app/routes/vault/cluster/secrets/backend/configuration/index.js

```diff
@@ -5,28 +5,132 @@
 
 import { service } from '@ember/service';
 import Route from '@ember/routing/route';
+import { CONFIGURABLE_SECRET_ENGINES } from 'vault/helpers/mountable-secret-engines';
+import { allEngines } from 'vault/helpers/mountable-secret-engines';
+import { hash } from 'rsvp';
+/**
+ * This route is responsible for fetching all configuration model(s).
+ * This includes the mount-configuration model attached to the secret-engine model via a belongsTo relationship.
+ * As well as any additional configuration models if the engine is a configurable engine.
+ */
 
 export default class SecretsBackendConfigurationRoute extends Route {
   @service store;
 
   async model() {
-    const backend = this.modelFor('vault.cluster.secrets.backend');
-    if (backend.isV2KV) {
+    const secretEngineModel = this.modelFor('vault.cluster.secrets.backend');
+    if (secretEngineModel.isV2KV) {
```

Comment:
nit: it might be helpful to move all this logic into a separate function that modifies the passed `secretEngineModel` and returns it with the modifications, so that the `model` function is easier to read. It's not ideal for methods to mutate the things it's passed and then pass it back, but I think it would make a big improvement for legibility.

Reply:
Agree. And ultimately the KVv2 logic here should be improved. I've avoided touching anything not AWS / SSH related all for scope reasons. I'm going to create a nice-to-have ticket for this epic to clean up the KVv2 work. However, just in case I do something breaking to KVv2 I'm going to add a link to this comment and put that in the JIRA ticket.

Reply:
udpate: ended up removing mirage from this tests and testing actual api.